### PR TITLE
Implement resolution

### PR DESCRIPTION
**Pull request by jankupis**

### CODE REVIEW

File: Resolution/Resolution.Tests/Clauses/ClauseCollectionBuilderTests.cs [hunk 1]
```diff
@@ -27,10 +27,10 @@ public void BuildsCorrectClauseCollection()
...
             Assert.AreEqual(1, clauseCollection[1].PositiveLiterals.Count);
-            Assert.AreEqual(literal2, clauseCollection[1].PositiveLiterals[0]);
+            Assert.IsTrue(clauseCollection[1].PositiveLiterals.Contains(literal2));
 
             Assert.AreEqual(1, clauseCollection[1].NegativeLiterals.Count);
-            Assert.AreEqual(literal1, clauseCollection[1].NegativeLiterals[0]);
+            Assert.IsTrue(clauseCollection[1].NegativeLiterals.Contains(literal1));
         }
     }
 }
```
Comment (debskipawel) [inline Resolution/Resolution.Tests/Clauses/ClauseCollectionBuilderTests.cs]: No newline at the end of the file, how would it look if I tried to cat in on my terminal?

File: Resolution/Resolution.Tests/VisitorsTests/ClauseMakerVisitorTests.cs [hunk 2]
```diff
@@ -59,15 +59,15 @@ public void ConjunctionOfClausesTest()
...
 
             Assert.AreEqual(1, clauseCollection[1].PositiveLiterals.Count);
-            Assert.AreEqual(s, clauseCollection[1].PositiveLiterals[0]);
+            Assert.IsTrue(clauseCollection[1].PositiveLiterals.Contains(s));
             Assert.AreEqual(1, clauseCollection[1].NegativeLiterals.Count);
-            Assert.AreEqual(t, clauseCollection[1].NegativeLiterals[0]);
+            Assert.IsTrue(clauseCollection[1].NegativeLiterals.Contains(t));
         }
     }
 }
```
Comment (debskipawel) [inline Resolution/Resolution.Tests/VisitorsTests/ClauseMakerVisitorTests.cs]: No newline at the end of the file, how would it look if I tried to cat in on my terminal?

**Review by debskipawel — CHANGES_REQUESTED** [review]

File: Resolution/Resolution.Tests/VisitorsTests/ClauseMakerVisitorTests.cs (hunk 2 above)
Comment (ArturChmura) [reply Resolution/Resolution.Tests/VisitorsTests/ClauseMakerVisitorTests.cs]: like that

**Review by debskipawel — APPROVED** [review]
Supi

**Pull request merged**